File: VolleyLib/src/main/java/com/kronos/volley/download/adapter/DataObservable.java
package com.kronos.volley.download.adapter;

import android.database.Observable;

/**
 * Created by Leif Zhang on 16/8/18.
 * Email leifzhanggithub@gmail.com
 */
public class DataObservable extends Observable<IObserver> {
    public void notifyChanged() {
        synchronized (mObservers) {
            for (int i = mObservers.size() - 1; i >= 0; i--) {
                mObservers.get(i).onChanged();
            }
        }
    }
}

File: DownloadLib/src/main/java/com/kronos/download/adapter/DataObservable.kt
package com.kronos.download.adapter

import android.database.Observable

/**
 * Created by Leif Zhang on 16/8/18.
 * Email leifzhanggithub@gmail.com
 */
class DataObservable : Observable<IObserver>() {
    fun notifyChanged() {
        synchronized(mObservers) {
            for (i in mObservers.indices.reversed()) {
                mObservers[i].onChanged()
            }
        }
    }

}
